usr/share/rear/restore/default/900_create_missing_directories.sh

```diff
@@ -1,38 +1,49 @@
 #
-# some backup SW doesn't restore mountpoints properly
+# Create missing directories.
+# For background information and reasoning see
+# https://github.com/rear/rear/issues/1455#issuecomment-324904017
+# that reads:
+#   Typical backup software assumes that the restore always happens
+#   on top of a system that was installed the traditional way,
+#   so that for the backup software it is safe to assume that
+#   those directories are present. From the perspective
+#   of the backup software there is therefore no need
+#   to include such directories in a full backup.
+#   Bottom line is, it is the duty of ReaR to make sure
+#   that these special directories really do exist.
 #
-#
-# create missing directories
-pushd $TARGET_FS_ROOT >/dev/null
-if [[ -f "$VAR_DIR/recovery/mountpoint_permissions" ]] ; then
-    LogPrint "Restore the Mountpoints (with permissions) from $VAR_DIR/recovery/mountpoint_permissions"
-    while read _dir mode userid groupid
-    do
-        ! [[ -d "$_dir" ]] && mkdir -p $_dir
-        chmod $mode $_dir
-        chown $userid:$groupid $_dir
-    done < <(cat "$VAR_DIR/recovery/mountpoint_permissions")
-elif [[ -z "$MOUNTPOINTS_TO_RESTORE" ]] ; then
-    # keep this for backward compatibility
-    mkdir -p mnt proc run sys tmp dev/pts dev/shm
-else
-    # keep this for backward compatibility
-    LogPrint "Restore the Mountpoints $MOUNTPOINTS_TO_RESTORE".
-    mkdir -p $MOUNTPOINTS_TO_RESTORE
-    # ensure some important mountpoints will be restored:
-    for _dir in mnt proc sys tmp dev/pts dev/shm
-    do
-      ! [ -d "$_dir" ] && mkdir -p $_dir
-    done
+pushd $TARGET_FS_ROOT 1>/dev/null
+
+local tmp_directories="tmp var/tmp"
+# First of all some generic directories that are created in any case:
+for directory in mnt proc run sys dev/pts dev/shm $tmp_directories ; do
+    test -d "$directory" || mkdir $v -p $directory 1>&2
+done
+# Set permissions for 'tmp' directories (cf. issue #1455):
+for tmp_dir in $tmp_directories ; do
+    chmod $v 1777 $tmp_dir 1>&2
```

Comment:
What about saving original /tmp and /var/tmp permissions to _mountpoint_permissions_ instead of blindly forcing 1777 ?
I know it is kind of standard to have temp dirs with this set of permissions but we are trying to get system closest to original as possible, right?

Reply:
I have issues with the naming:
I won't expect /tmp and /var/tmp in a mountpoint_permissions file.
In other words:
In general I agree with better support for permissions c.f.
https://github.com/rear/rear/issues/1455#issuecomment-324949532
but currently I do not want to overhaul that whole functionality.